lib/AArch64Base.ml

```diff
@@ -595,6 +599,16 @@ let type_reg r =
   | Vreg  (_,(n_elt,sz)) -> Array (TestType.tr_nbits sz,n_elt)
   | _ -> Base "int"
 
+let is_sysreg = function
+  | SysReg _ -> true
+  | _ -> false
+
+let is_spsysreg = function
+  | SysReg r -> is_special_purpose r
+  | _ -> false
+
+let is_non_sp_sysreg r = is_sysreg r && not (is_spsysreg r)
```

Comment:
This is not necessary and since this is not supported by other architectures, let derived where it is needed.